### PR TITLE
Integrate Supabase for Article Fetching, Update Project Images, and Add Social Sharing Functionality

### LINKED ISSUE #158: Fetch Articles Data from Supabase Database
**Description**

We need to fetch articles data from our Supabase database to display on our frontend. The data should include the following fields:

- Article Title
- Article Category
- Photos
- Article Description

**Tasks**

**Database Schema:**

- Ensure that the Supabase table (articles) has the necessary columns:

- title (string)
- category (string)
- photos (array of strings or a JSON object, depending on how you store multiple photos)
- description (text or string)

**Fetch Data:**

- Create a function to fetch data from the articles table in Supabase.
- Handle the response to include the necessary fields.

**Error Handling:**

- Implement error handling to manage scenarios where data fetching fails.
- Display appropriate error messages to the user.

**State Management:**

- Use React state to store the fetched data and any errors.
- Update the state accordingly based on the fetch response.

### LINKED ISSUE #171: Bug Fix:   Resolve environment variables conflicting with expo router.
### Problem:

The routing system in the medica app is currently experiencing disruptions caused by incompatible environment variables. This issue manifests as:

Symptoms:

- Navigation between screens becomes inconsistent or non-functional.

- Users encounter unexpected redirects or errors when accessing different app sections.

### Steps to Reproduce:

- Navigate through different sections of the app.

The routing system should seamlessly navigate users between screens without interruptions or errors caused by environmental configuration issues.

### Tasks to Address:

- [x] Identify specific environment variables causing routing disruptions.

- [x] Implement adjustments to ensure compatibility with React Native Expo.

- [x] Test thoroughly to validate fixes and ensure smooth navigation across all app sections.

### Screen shot

Screenshot from 2024-06-25 11-29-40

### PR DESCRIPTION
**Pull request by munezeromicha**
**Overview**

This pull request introduces several significant enhancements to the article fetching and sharing functionalities in our application. The key features include fetching articles from the Supabase database, updating the article pictures, and enabling users to share articles via social media and messaging applications.

**Changes Implemented**

1. **Fetching Articles from Supabase Database**

- Integrated Supabase as the backend to fetch articles.
- Ensured secure and optimized data retrieval processes.

2. **Updating Article Pictures**

- Removed previously used images from the whole project.
- Implemented a new image handling system for updating project pictures.
- Ensured the new article images are properly stored and referenced in Supabase.
- Updated the frontend to display the new images correctly.

3. **Sharing Articles via Social Media and Messaging Applications**

- Added sharing functionality to articles, allowing users to share via popular social media platforms (Facebook, Twitter, LinkedIn, etc.) and messaging apps.
- Integrated share buttons and modals into the article UI.

4. **ScreenShoots:**

<div style="display: flex; flex-wrap: wrap;">
    Screenshot 1
    Screenshot 2
    Screenshot 3
    Screenshot 4
    Screenshot 5
    Screenshot 6
</div>

### CODE REVIEW

**Review by Irirwanirira — APPROVED** [review]
Good,

**Review by mr3nz1 — APPROVED** [review]

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+EXPO_PUBLIC_API_URL=http://api.local
+EXPO_PUBLIC_API_KEY=api_key
+
+SUPABASE_URL=https://vbwbfflzxuhktdvpbspd.supabase.co
+SUPABASE_NON_KEY=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZid2JmZmx6eHVoa3RkdnBic3BkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTgyNTg0MjgsImV4cCI6MjAzMzgzNDQyOH0.eOPZwYgyg3Ic-jbXW4Rehd2RDgqf-kt4vr30w_O9h3c
```
Comment (Nkbtemmy) [inline .env]: First of all; it is not allowed to push `.env` file

Comment (Nkbtemmy) [reply .env]: You provide a variable with not values in .env.example

File: .gitignore [hunk 2]
```diff
@@ -6,7 +6,7 @@ node_modules/
 .expo/
 dist/
 web-build/
-.env
+.env.example
```
Comment (Nkbtemmy) [inline .gitignore:10]: why did you change it; it was better how it was

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]
This PR has many files changed; Don't repeat it; 
changing Images should have been on separate PR !!

File: lib/supabase.ts [hunk 3]
```diff
@@ -2,9 +2,8 @@ import "react-native-url-polyfill/auto";
 import { createClient } from "@supabase/supabase-js";
 import AsyncStorage from "@react-native-async-storage/async-storage";
 
-export const supabase = createClient(
-  "https://vbwbfflzxuhktdvpbspd.supabase.co",
-  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZid2JmZmx6eHVoa3RkdnBic3BkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTgyNTg0MjgsImV4cCI6MjAzMzgzNDQyOH0.eOPZwYgyg3Ic-jbXW4Rehd2RDgqf-kt4vr30w_O9h3c",
+
+export const supabase = createClient('https://vbwbfflzxuhktdvpbspd.supabase.co', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InZid2JmZmx6eHVoa3RkdnBic3BkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTgyNTg0MjgsImV4cCI6MjAzMzgzNDQyOH0.eOPZwYgyg3Ic-jbXW4Rehd2RDgqf-kt4vr30w_O9h3c',
```
Comment (Nkbtemmy) [inline lib/supabase.ts]: This should be in .env

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]

File: lib/supabase.ts (hunk 3 above)
Comment (Nkbtemmy) [inline lib/supabase.ts]: Why Still here ??

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]

**Review by Nkbtemmy — APPROVED** [review]
